design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,471 @@
+The solution is to allow different deployments and behave differently based on the use-case chosen. The following decision tree depicts customer uses-cases for Secrets Provider deployments:
+
+ ![Secrets Provider flavors decision flow chart](https://user-images.githubusercontent.com/31179631/85747023-975bf500-b70f-11ea-8e26-1134068fe655.png) 
+
+These variations will be supported using the same Secrets Provider image, but the behavior will vary dynamically depending on the chosen deployment method that is configured via Helm Chart.
+
+*In this document we will focus on **Milestone 1***, deploying the Secrets Provider as a Job
+
+### Milestone 1: Serve K8s secrets to multiple applications once (no rotation)
+
+### Design
+
+Customer will install our **Helm Chart** to deploy the Secrets Provider as a **Job**. Once installed, the Job spins up and authenticates to Conjur/DAP via authn-k8s.
+It will then fetch all the K8s secrets update them with the Conjur secrets they require and terminate upon completion.
+
+#### How does a Job installed via Helm answer the requirements?
+
+Running as a **Job** allows separation from the applications' deployment and serve multiple applications at once. 
+
+Because for this Milestone we are concerned with updating K8s Secrets once at intial spin up, a **Job** is the most native solution. It will terminate upon task completion and not waste customer's resources.
+
+Kubernetes Helm is a tool that streamlines and organizes the management of Kubernetes installation and deployment processes. When we use Helm for deployment and lifecycle management, we can provide our customer's a one-click solution even though the way at which we deploy changes.
+
+#### What drawbacks does this solution have?
+
+**Job** is not an existing application identity granularity.
+
+ As a result, one can define the host representing the Secrets Provider in Conjur policy using only **namespace** with/without **service account** granularities.
+
+To handle the missing **Job** granularity there are 2 options:
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Keep using only **namespace** and/or **service account** granularities. | \- Since Secrets Provider runs as a separate entity, we suggest creating a dedicated service account for Secrets Provider.<br />That way, the **service account** granularity serves as an exact match and **Job** granularity is redundant<br />- No code changes | \- Service account might be reused, which allows other apps to authenticate using Secrets Provider host<br />- Break application identity convention | Free              |
```

Comment:
Can you elaborate on why it breaks application identity (a.k.a machine identity) convention?

Reply:
Because we currently have a Deployment application identity to represent the type deployment
 `kind: deployment`. We are suggesting that for Job, we don't have a Job granularity even though the secrets provider will be deployed as a job `kind: job`